docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
> So if I understand you properly, you're saying that my use-case number 4 is actually effectively impossible to support on clients, since the protocol doesn't have an affordance for that?

Not *impossible* per se, just that the effect it has is not what you'd expect. Suppose you configure a client to support TLSv1 and TLSv1.2 only. The dialog is like this:

1. Client sends Client Hello, and the version field in there is set to the highest supported version (TLSv1.2).
2. The server sends back a Server Hello, selecting the highest version both it and the client support. The server only supports up to TLSv1.1, so it sends back TLSv1.1.

Now, the effect of "skip intermediate versions" is that this handshake will fail, because the server chose TLSv1.1 which the client doesn't want to support. This is *in spite* of the fact that both the client and server may have a supported common protocol in TLSv1! The client will just say "welp didn't support that protocol version, raise an error". That's a pretty stupid failure mode when in fact the two endpoints *can* communicate.

As to your proposed argument names, that's fine by me.